faculty/__init__.py

```diff
@@ -17,7 +17,85 @@
 import faculty.clients
 
 
-def client(resource, **kwargs):
-    session = faculty.session.get_session(**kwargs)
+def client(
+    resource,
+    credentials_path=None,
+    profile_name=None,
+    domain=None,
+    protocol=None,
+    client_id=None,
+    client_secret=None,
+    access_token_cache=None,
+):
+    """
+    Construct a client for a Faculty resource.
+
+    Parameters
+    ----------
+    resource : str
+        The resource to construct a client for.
+    credentials_path : str or pathlib.Path, optional
+        The path of a file to load Faculty credentials from.
+    profile_name : str, optional
+        The name of the profile to read from the credentials file. The default
+        is 'default'.
+    domain : str, optional
+        The domain to access Faculty services at. This is provided when you
+        generate credentials in the platform.
+    protocol : str, optional
+        Either 'http' or 'https' (the default).
+    client_id : str, optional
+        The Faculty client ID to user. This is provided when you generate
+        credentials in the platform.
+    client_secret : str, optional
+        The Faculty client secret to use. This is provided when you generate
+        credentials in the platform.
+    access_token_cache : faculty.session.accesstoken.AccessTokenMemoryCache or
+    faculty.session.accesstoken.AccessTokenMemoryCache, optional
+        Set the access token cache used. The default is an
+        AccessTokenMemoryCache.
+
+    Examples
+    --------
+    To construct a client loading default configuration and credentials (from
+    ~/.config/faculty/credentials or from environment variables), simply call
+    this function with the resource type as a string:
+
+    >>> faculty.client("account")
+    <faculty.clients.account.AccountClient object at 0x10d4b7fd0>
+
+    To load a different configuration file, or load a profile from the file
+    other than 'default', pass the `credentials_path` or `profile_name`
+    arguments respectively:
+
+    >>> faculty.client(
+    ...     "account",
+    ...     credentials_path="other/credentials",
+    ...     profile_name="custom",
+    ... )
+    <faculty.clients.account.AccountClient object at 0x10e447550>
+
+    To override any of the domain, protocol, client ID or client secret read
+    from a credentials file or from environment variables, pass their
+    respective arguments:
+
+    >>> faculty.client(
+    ...     "account",
+    ...     domain="services.faculty.myorganisation.com",
+    ...     protocol="http",
+    ...     client_id="d047dad1-175b-4810-84b1-c0ff6bbcf456",
+    ...     client_secret="Vk4a1FgSKlTplkK0GkToAdRTdsoU4XHuwCMQ",
+    ... )
+    <faculty.clients.account.AccountClient object at 0x10e4472b0>
+    """
```

Comment:
Should we mention what happens if you pass both `credentials_path/profile_name` *and* `domain/protocol/client_id/client_secret`?

Reply:
I had covered this, but it wasn't super clear. Have reworded to clarify.